src/writer/xapianIndexer.cpp

```diff
@@ -83,7 +98,16 @@ void XapianIndexer::indexingPrelude()
   writableDatabase.begin_transaction(true);
 }
 
-void XapianIndexer::indexTitle(const std::string& path, const std::string& title)
+/*
+ * TODO:
+ * Currently for title index we are storing path twice (redirectPath/path in
+ * valuesmap and path in index data) to secure backward compatibility with older
+ * kiwixes that uses data in the index to load the articles. In the future, we
+ * might want to keep only one of these to reduce index size while supporting
+ * the collape on path feature.
+ */
```

Comment:
In fact, this is not only backward compatibility.
We are still getting a entry using the path stored in the data
https://github.com/openzim/libzim/blob/master/src/search_internal.h#L88

And it is a "good" thing. The data of the document is the id to get the entry in the zim file.
(If we change this, we may store the id (number) of the entry instead of the path. But it is not so easy to do)

The "path" we are storing in the values is something different, it is not the id of the entry, it is a collapsing value and it is normal it is different.

Reply:
Oh! that means it will be inevitable to have paths stored in two places until we find another **efficient** method to implement the collapse feature. I should change the comment then.